src/model/guild/member.rs

```diff
@@ -355,7 +355,7 @@ impl Member {
     ///
     /// [`kick`]: #method.kick
     #[cfg(feature = "http")]
-    pub fn kick_with_reason(&self, cache_http: impl CacheHttp, reason: Option<&str>) -> Result<()> {
+    pub fn kick_with_reason(&self, cache_http: impl CacheHttp, reason: &str) -> Result<()> {
```

Comment:
Alright, you have almost got my all of wishes, but you seem to have missed one from my last review. 
I wanted for you administer this name change to all of the other `kick` methods as well, so that we wouldn't break the higher-level interface people would most likely use.

What I mean is that, for each `kick` method, there should be an accompanying `kick_with_reason` that the former would use. In the same way you have done it with `Member::kick`.